modules/dcache-chimera/src/main/java/org/dcache/chimera/namespace/ChimeraCleaner.java

```diff
@@ -468,22 +468,26 @@ private void runNotification()
                 "SELECT ipnfsid FROM t_locationinfo_trash t1 " +
                 "WHERE itype=2 AND NOT EXISTS (SELECT 1 FROM t_locationinfo_trash t2 WHERE t2.ipnfsid=t1.ipnfsid AND t2.itype <> 2) LIMIT=?";
 
-        List<String> queryList = new ArrayList<String>();
+        List<String> queryFail = new ArrayList<String>();
         do {
-            queryList = _db.queryForList(QUERY, String.class, QUERYLIMIT);
-
-            if( queryList.isEmpty()) {
-                break;
             }
+            List<String> queryList = _db.queryForList(QUERY, String.class, QUERYLIMIT);
 
             for (String id : queryList) {
                 try {
                     sendDeleteNotifications(new PnfsId(id)).get();
                     _db.update("DELETE FROM t_locationinfo_trash WHERE ipnfsid=? AND itype=2", id);
                 } catch (ExecutionException e) {
+                    if(!queryFail.contains(id)) {
+                        queryFail.add(id);
+                    }
                     _log.warn(e.getCause().getMessage());
                 }
             }
+            if(queryFail.size() >= QUERYLIMIT){
+                _log.warn("Error: Too many unresolved IDs to finish the loop.");
```

Comment:
Use of log level warning and error in the start of log message is contradictory. 

Also improve the log message,   "IDs to finish the loop." is not a very useful error message.